### DISCUSSION TITLE
Prisma generates a query that should return results, but returns undefined

### DISCUSSION THREAD

**Discussion by alexmanzo** [prisma/prisma#6797] [Q&A] [answered]
Hey all! I've been working on implementing Prisma into a codebase that, until this point, has depended on raw SQL queries.

I'm running into an issue right now where I'm trying to access from pretty deeply nested relations. I'm leveraging the Fluent API, but struggling because the following ends up with `results` being `undefined`.

Due to security/privacy, I've replaced actual names with `model1`, `model2`, etc. Sorry in advance for the confusion.

```js
let results = await prisma.model1
  .findUnique({ where: { id: entityId } })
  .model2()
  .model3()
  .model4({ where: { isActive: 1 } })
  .model5()
  .model6({
    select: {
      label: true,
      id: true
    }
  });
  ```
Through logging, I also the SQL query which I ran in pgAdmin with the appropriate values. That returned the results I would expect.
  ```sql
  SELECT "public"."model6"."id", "public"."model6"."name" FROM "public"."model6" WHERE "public"."model6"."id" IN ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12) OFFSET $13
  ```

Can anyone shed light on how Prisma is successfully constructing the query with the correct values, but not returning results?

Models:

```prisma
model model1 {
  // ....
  model2 model2[]

  @@unique([groupId, name], name: "model1_c01")
}

model model2 {
  model1Id    Int         @map("model1_id")
  model3Id    Int         @map("model3_id")
  model1      model1      @relation(fields: [model1Id], references: [id])
  model3      model3      @relation(fields: [model3Id], references: [id])

  @@id([model1Id, model3Id])
}

model model3 {
  // ....
  model4      model4[]      
  model2      model2[]      
}

model model4 {
  // ....
  model3Id            Int?                    @map("model3_id")
  model5              model5[]
}

model model5 {
  model4Id    Int         @map("model4_id")
  model6Id    Int         @map("model6_id")
  model4      model4      @relation(fields: [model4Id], references: [id])
  model6      model6      @relation(fields: [model6Id], references: [id])

  @@id([model4Id, model6Id])
  @@unique([model6Id, model4Id], name: "model5_c01")
  @@index([model4Id], name: "model5_i01")
  @@index([model6Id], name: "model5_i02")
}

model model6 {
  // ....
  model5 model5[]
}

```

**Comment by janpio** [comment, marked as answer]
That sounds very much like a bug. Can you create an issue so we can investigate this? (Optimally provide a small reproduction repository that works almost out of the box with schema and code). Thanks.

**Reply by alexmanzo** [reply]
Yep! Thanks for touching base and I'll try to get a small reproduction repo up for it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
